src/overlays/actors/ovl_En_Sc_Ruppe/z_en_sc_ruppe.c

```diff
@@ -31,29 +37,155 @@ const ActorInit En_Sc_Ruppe_InitVars = {
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(globalCtx, this, 32.0f, 30.0f, 0.0f, 4U);
+}
+
+s32 func_80BD697C(s16 this) {
+    switch (this) {
+        case 0:
+            if (gSaveContext.save.weekEventReg[53] & 4) {
+                gSaveContext.save.weekEventReg[53] &= (u8)~4;
+                return 1;
+            }
+            break;
+        case 1:
+            if (gSaveContext.save.weekEventReg[53] & 0x80) {
+                gSaveContext.save.weekEventReg[53] &= (u8)~0x80;
+                return 1;
+            }
+            break;
+        case 2:
+            if (gSaveContext.save.weekEventReg[54] & 1) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~1;
+                return 1;
+            }
+            break;
+        case 3:
+            if (gSaveContext.save.weekEventReg[54] & 2) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~2;
+                return 1;
+            }
+            break;
+        case 4:
+            if (gSaveContext.save.weekEventReg[54] & 4) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~4;
+                return 1;
+            }
+            break;
+        case 5:
+            if ((gSaveContext.save.weekEventReg[54] & 8)) {
+                gSaveContext.save.weekEventReg[54] &= (u8)~8;
+                return 1;
+            }
+            break;
+    }
+    return 0;
+}
+
+void func_80BD6A8C(EnScRuppe* this, GlobalContext* globalCtx) {
+    if (this->collider.base.ocFlags1 & 2) {
+        this->rupeeDisplayTime = 0;
+        this->actor.gravity = 0.0f;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_GET_RUPY);
+        func_801159EC(sRupeeInfo[this->rupeeIndex].amount);
+        this->actionFunc = func_80BD6B18;
+    }
+    this->actor.shape.rot.y += 0x1F4;
+    Actor_MoveWithGravity(&this->actor);
+}
+
+void func_80BD6B18(EnScRuppe* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (this->rupeeDisplayTime > 30) {
+        if (func_80BD697C(this->rupeeIndex)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        f32 scale;
+        this->rupeeDisplayTime++;
+        this->actor.world.pos = player->actor.world.pos;
+        this->actor.world.pos.y += 40.0f;
+        scale = (30.0f - this->rupeeDisplayTime) * 0.001f;
+        Actor_SetScale(&this->actor, scale);
+    }
+    this->actor.shape.rot.y += 0x3E8;
+    Actor_MoveWithGravity(&this->actor);
+}
+
+void EnScRuppe_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderCylinder* collider;
+    EnScRuppe* this = THIS;
+
+    collider = &this->collider;
+    Collider_InitCylinder(globalCtx, collider);
+    Collider_InitAndSetCylinder(globalCtx, collider, &this->actor, &sCylinderInit);
+    Actor_SetScale(&this->actor, 0.03f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 10.0f);
+    this->rupeeIndex = this->actor.params & 31;
```

Comment:
```suggestion
    this->rupeeIndex = this->actor.params & 0x1F;
```
I think we generally keep params in hex when bitwise AND-ing as well as making this a `#define` in the header.